File: SavingsAccount.java
public class SavingsAccount extends BankAccount{

    //a subclass of BankAccount

        private double interest;

        public SavingsAccount(String n, double b, double intRate) {
            super(n,b);
            interest = intRate;

        }

        //a method to add an amount of to the balance in a BankAccount
        public void addInterest(){
            double add;
            add = (balance/100) * interest;
            balance += add;
        }

    }

File: SavingsAccount.cpp
//
//  SavingsAccount.cpp
//  BankAccount
//
//  Created by Rebekah on 11/20/18.
//  Copyright © 2018 Rebekah. All rights reserved.
//

#include "SavingsAccount.hpp"
#include "BankAccount.hpp"
#include "CheckingAccount.hpp"

/*
 a constructor for SavingAccount
 */
SavingsAccount::SavingsAccount(string n, double b, double i){
    this -> name = n;
    this -> balance = b;
    this -> interestRate = i;
}

/*
 adds interest to a SavingsAccount
 */
void SavingsAccount::addInterest(){
    this->deposit((this->getBalance()/100)*interestRate);
}
